libcodechecker/libclient/client.py

```diff
@@ -29,7 +32,20 @@ def check_api_version(client):
     """
 
     version = client.getAPIVersion()
-    supp_major_version = SUPPORTED_VERSION.split('.')[0]
+    supp_major_version = SUPPORTED_RUN_API_VERSION.split('.')[0]
+    api_major_version = version.split('.')[0]
+
+    # There is NO compatibility between major versions.
+    return supp_major_version == api_major_version
+
+
+def check_product_api_version(client):
+    """
+    Check if server product API is supported by the client.
+    """
+
+    version = client.getAPIVersion()
```

Comment:
I would, yes. It could very well be that the run API changes earlier than the product API. They are very essentially two different systems, which could have two different clients, etc.